codec/reflectcodec/type_codec.go

```diff
@@ -210,23 +208,67 @@ func (c *genericCodec) size(value reflect.Value) (int, bool, error) {
 		return size, constSize, nil
 
 	case reflect.Map:
-		size := wrappers.IntLen
-		constSize := true
-		// add the length of each key and value
-		for _, key := range value.MapKeys() {
-			innerSize, _, err := c.size(key)
-			if err != nil {
-				return 0, false, err
+		iter := value.MapRange()
```

Comment:
This was switched to use `MapRange` so that `Size` is `O(1)` if the key and value are constant sized.